src/NativeScript/inspector/SuppressAllPausesEnabler.cpp

```diff
@@ -0,0 +1,31 @@
+#include "SuppressAllPausesEnabler.h"
+#include <JavaScriptCore/Debugger.h>
+
+namespace NativeScript {
+
+SuppressAllPausesEnabler::SuppressAllPausesEnabler(JSC::JSGlobalObject& globalObject)
+    : m_globalObject(globalObject) {
+    if (this->suppressAllPauses()) {
+        if (JSC::Debugger* debugger = this->m_globalObject.debugger())
+            debugger->setSuppressAllPauses(true);
+    }
+}
+
+SuppressAllPausesEnabler::~SuppressAllPausesEnabler() {
+    if (this->suppressAllPauses()) {
+        if (JSC::Debugger* debugger = this->m_globalObject.debugger()) {
+            debugger->setSuppressAllPauses(false);
+        }
+    }
+}
+
+bool SuppressAllPausesEnabler::suppressAllPauses() {
```

Comment:
This will work better as a static method. The environment variable is global so I don't think it makes sense to update a class field with its value.